### DISCUSSION TITLE
Vue 3 Discussion - Vue Stripe Version 5

### DISCUSSION THREAD

**Discussion by jofftiquez** [jofftiquez/vue-stripe-checkout#236] [General]
## Vue 3 and Version 5 Discussion

This is a continuation of https://github.com/vue-stripe/vue-stripe/issues/148. I decided to move the discussion here for a better forum-like experience. 

## Today

The latest (3rd) iteration of the vue 3 version is available here https://github.com/vue-stripe/vue-stripe/tree/refactor/next

New docs here https://github.com/vue-stripe/docs 

The major changes here are as follows:

- It now uses vue-demi
- Better structuring of core composables - so that they can be used as standalone utilities
- Upgrade component structure to renderless component
- New docs using vite press see early example https://vue-stripe-docs-ossph.vercel.app/

**Comment by jofftiquez** [comment]
Will close this discussion in favor of https://github.com/orgs/vue-stripe/projects/1/views/1

**Comment by targetlucked69** [comment]
Here are better alternatives that works with Vue 3:

- https://github.com/ectoflow/vue-stripe-js
- https://github.com/frandiox/vue-use-stripe

**Reply by dapug** [reply]
What makes either of these "better"?  Simply that they are actually implemented and working?

**Comment by piotrczyz** [comment]
Is there any other discussion or is it just a dead discussion?

**Comment by jofftiquez** [comment]
I am currently testing which platform is better for discussion issues, obviously, this GitHub Discussion hasn't been maintained for a while but we can revive this or check the poll here: https://twitter.com/jrtiquez/status/1686812250190082049

If you can't see the link please join the Twitter community first: https://twitter.com/i/communities/1506778375733403648

**Reply by douglasg14b** [reply]
Discord is where information goes to die, please no. Almost every discord server for a FOSS project just turns into repeatave Q&A that sucks the life out of maintainers, and devolves into a graveyard "1000 threads, 30 answers". Where as github discussions & issues are actually indexed & reachable from the wider internet, and promote long-term discussions.

This thread is a good example, it's 3 years old. Information on discord dies rapidly, none of this would be discoverable.

**Comment by jofftiquez** [comment]
Current progress as of Jan. 23, 2024

I made another iteration to the current version 5 branch. The reason is that I wanted to use vue-demi to support vue 2 as well. The current working branch is here https://github.com/vue-stripe/vue-stripe/tree/refactor/next. 

A lot of core changes has been made, specially how composables are structured. It is now decoupled to the component so we can use it as a standalone utility if you want to implement your own version of Element or if just wanna use the Stripe instance altogether. 

A new doc using VitePress is also available at https://github.com/vue-stripe/docs. I think this will be the best option as I can use the components directly in the docs for live demo which will save me a lot of time writing external demo repositories.

**Comment by jofftiquez** [comment]
Hi all, a preview of the new documentation site is now available in this temporary link. 

https://vue-stripe-docs-next--pr355-refactor-next-9qiddyov.web.app/

See PR #355 

Please let me know if you have any comments regarding the new documentation. Thank you!

**Reply by dataexcess** [reply]
Amazing! Does this mean vue-stripe for vue 3 is production ready?

**Reply by hackel** [reply]
Please consider adding a Migration page to the docs.

It wasn't immediately obvious how to replace the `StripeElementCard` component.

**Reply by jofftiquez** [reply]
> Amazing! Does this mean vue-stripe for vue 3 is production ready?

@dataexcess almost there! Still doing a few testing now

**Reply by jofftiquez** [reply]
> Please consider adding a Migration page to the docs.
> 
> It wasn't immediately obvious how to replace the `StripeElementCard` component.

Good point @hackel thanks.

**Reply by dataexcess** [reply]
> > Amazing! Does this mean vue-stripe for vue 3 is production ready?
> 
> @dataexcess almost there! Still doing a few testing now

Hows the testing going? ^^ Have a project coming up in a few months and would love to use this! Is there a rough ETA? thank you and great work!

**Reply by dataexcess** [reply]
so I guess the project is paused? none of the posted docs urls are active anymore...

**Comment by hackel** [comment]
I tried to install the `refactor/next` branch, but get a dependency error with @vue/composition-api:
```
npm error Conflicting peer dependency: vue@3.5.11
npm error node_modules/vue
npm error   peer vue@">= 2.5 < 2.7" from @vue/composition-api@1.7.2
npm error   node_modules/@vue/composition-api
npm error     peerOptional @vue/composition-api@"^1.7.2" from @vue-stripe/vue-stripe@5.0.0
npm error     node_modules/@vue-stripe/vue-stripe
npm error       @vue-stripe/vue-stripe@"vue-stripe/vue-stripe#refactor/next" from the root project
```

**Reply by jofftiquez** [reply]
How did you install it?

**Comment by jofftiquez** [comment]
Hi all, version 5 is now live. Let me know if you tried it already. Thanks!
